x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/context.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { createContext, memo, useContext, useEffect, useMemo, useState } from 'react';
+import { useAssistantContext } from '@kbn/elastic-assistant';
+import type { AttackDiscoveryAlert } from '@kbn/elastic-assistant-common';
+import type { AttackDetailsProps } from './types';
+import { FlyoutLoading } from '../shared/components/flyout_loading';
+import { FlyoutError } from '../shared/components/flyout_error';
+import { useFindAttackDiscoveries } from '../../attack_discovery/pages/use_find_attack_discoveries';
+
+export interface AttackDetailsContext {
+  /**
+   * Id of the attack document
+   */
+  attackId: string;
+  /**
+   * Attack details
+   */
+  attack: AttackDiscoveryAlert;
+}
+
+/**
+ * A context provider for Attack Details flyout
+ */
+export const AttackDetailsContext = createContext<AttackDetailsContext | undefined>(undefined);
+
+export type AttackDetailsProviderProps = {
+  /**
+   * React components to render
+   */
+  children: React.ReactNode;
+} & Partial<AttackDetailsProps['params']>;
+
+export const AttackDetailsProvider = memo(({ attackId, children }: AttackDetailsProviderProps) => {
+  const { assistantAvailability, http } = useAssistantContext();
+
+  const { data, isLoading: loading } = useFindAttackDiscoveries({
+    ids: [attackId || ''],
+    http,
+    isAssistantEnabled: assistantAvailability.isAssistantEnabled,
+    page: 1,
+    perPage: 1,
```

Comment:
And if I'm not mistaken you should be able to remove this one at all. While the default value is 10 in the hook, I don't think we care as we're always fetching a single id. Let's remove it to keep the code as simple as possible